File: WebApi/Interfaces/ITrBoxService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrBoxDto, CreateTrBoxDto, UpdateTrBoxDto } from '../Models/TrBoxDtos';

export interface ITrBoxService {
  getAll(): Promise<ApiResponse<TrBoxDto[]>>;
  getById(id: number): Promise<ApiResponse<TrBoxDto>>;
  getByImportLineId(importLineId: number): Promise<ApiResponse<TrBoxDto[]>>;
  getByBoxNumber(boxNumber: string): Promise<ApiResponse<TrBoxDto[]>>;
  getActive(): Promise<ApiResponse<TrBoxDto[]>>;
  getByDescription(description: string): Promise<ApiResponse<TrBoxDto[]>>;
  create(createDto: CreateTrBoxDto): Promise<ApiResponse<TrBoxDto>>;
  update(id: number, updateDto: UpdateTrBoxDto): Promise<ApiResponse<TrBoxDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}

File: Interfaces/ITrBoxService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrBoxService
    {
        Task<ApiResponse<IEnumerable<TrBoxDto>>> GetAllAsync();
        Task<ApiResponse<TrBoxDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrBoxDto>>> GetByImportLineIdAsync(long importLineId);
        Task<ApiResponse<IEnumerable<TrBoxDto>>> GetByBoxNumberAsync(string boxNumber);
        Task<ApiResponse<IEnumerable<TrBoxDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<TrBoxDto>>> GetByDescriptionAsync(string description);
        Task<ApiResponse<TrBoxDto>> CreateAsync(CreateTrBoxDto createDto);
        Task<ApiResponse<TrBoxDto>> UpdateAsync(long id, UpdateTrBoxDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}